src/gt4py/next/iterator/embedded.py

```diff
@@ -1284,7 +1280,7 @@ def _tuple_assign(field, value, indices):
             _tuple_assign(f, v, indices)
     else:
         assert is_located_field(field)
-        field[indices] = value
+        field.field_setitem(indices, value)
```

Comment:
```suggestion
        field.field_setitem(named_indices, value)
```
Those are `named_indices` now right? I would rename the variable then. How about the `_build_tuple_result` function above? Is the `__getitem__` access correct there?

Reply:
yes, looks like there was a bug, not sure why we didn't run into it.